src/web/wms/src/main/java/org/geoserver/wms/web/data/ExternalGraphicPanel.java

```diff
@@ -350,7 +330,28 @@ private void updateVisibility(boolean b) {
         show.setVisible(!b);
     }
 
-    private boolean isUrlOrFile(final String uri) {
-        return externalURIPattern.matches(uri);
+    private boolean isUrl(final String uri) {
+        return uri.startsWith("http");
+    }
+
+    private File findResourceinStylesDirectory(WorkspaceInfo wsInfo, String value)
```

Comment:
This method is not needed, the logic is already provided in the `GeoServerDataDirectory` class:

A couple things - we don't want a File, we just want to check if a file exists.

The class `GeoServerDataDirectory` already has tested methods for these checks:
```
Resource icon = geoServerDataDirectory.getStylesget(value );
return icon.getType() == Resource.Type.RESOURCE;
```

Reply:
this method is basically wrapping the re-usable code, which is currently duplicated, perhaps the name of the method is misleading